File: BusinessObjects/Entities/ElementItem.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace forCrowd.Backbone.BusinessObjects.Entities
{
    public class ElementItem : BaseEntity
    {
        public ElementItem()
        {
            ElementCellSet = new HashSet<ElementCell>();
            ParentCellSet = new HashSet<ElementCell>();
        }

        public int Id { get; set; }

        public int ElementId { get; set; }

        [Required]
        [StringLength(150)]
        public string Name { get => name; set => name = value.Trim(); }

        public Element Element { get; set; }
        public ICollection<ElementCell> ElementCellSet { get; set; }
        [InverseProperty("SelectedElementItem")]
        public ICollection<ElementCell> ParentCellSet { get; set; }

        string name;

        public ElementCell AddCell(ElementField field)
        {
            Framework.Validations.ArgumentNullOrDefault(field, nameof(field));

            if (ElementCellSet.Any(item => item.ElementField == field))
                throw new System.Exception("An element item can't have more than one cell for the same field.");

            var cell = new ElementCell
            {
                ElementField = field,
                ElementItem = this
            };
            field.ElementCellSet.Add(cell);
            ElementCellSet.Add(cell);
            return cell;
        }
    }
}

File: AngularClient/src/main/core/entities/element-item.ts
import { EntityBase } from "./entity-base";
import { Element } from "./element";
import { ElementCell } from "./element-cell";
import { ElementFieldDataType } from "./element-field";

export class ElementItem extends EntityBase {

  // Server-side
  Id = 0;
  Element: Element;
  ElementCellSet: ElementCell[];
  ParentCellSet: ElementCell[];

  get Name(): string {
    return this.fields.name;
  }
  set Name(value: string) {
    this.fields.name = value.trim();
  }

  private fields: {
    name: string,
  } = {
      name: "",
    };

  initialize(): void {
    if (this.initialized) return;

    super.initialize();

    // Cells
    this.ElementCellSet.forEach(cell => {
      cell.initialize();
    });
  }
}
